movies/src/components/addMovie/AddMovie.js

```diff
@@ -1,71 +1,174 @@
-  const [isMovieAction, isMovieActionPopup] = useState(false);
-  const { handleClose } = props;
+const matchDispatchToProps = { updateMovies: addMovie };
 
-  const handleSubmit = (e) => {
-    e.preventDefault();
-    isMovieActionPopup(true);
-  };
+const AddMovie = ({ updateMovies, handleClose }) => {
+  const [isMovieAction, isMovieActionPopup] = useState(false);
 
   return (
     <>
       <CloseButton handleClose={handleClose} />
-      <form onSubmit={handleSubmit}>
-        <p className="modal-title">Add Movie</p>
-        <Input
-          id="movie-title"
-          title="Title"
-          labelClassName="input-label"
-        />
-        <Datepicker
-          id="movie-release"
-          title="Release date"
-          labelClassName="input-label"
-          value={this}
-        />
-        <Input
-          id="movie-url"
-          title="Movie url"
-          labelClassName="input-label"
-        />
-        <div className="modal-dropdown">
-          <span className="modal-label-text">Genre</span>
-          <MultiSelectInput options={Constants.GENRE} />
-        </div>
-        <Input
-          id="movie-overview"
-          title="overview"
-          labelClassName="input-label"
-        />
-        <Input
-          id="movie-runtime"
-          title="runtime"
-          labelClassName="input-label"
-        />
-        <div className="add-movie-buttons">
-          <Button
-            className="button button_secondary"
-            title="RESET"
-            handleClick={() => {}}
-          />
-          <Button
-            className="button button_primary"
-            title="SUBMIT"
-            type="submit"
-            handleClick={() => {}}
-          />
-        </div>
-      </form>
+      <p className="modal-title">Add Movie</p>
+      <Formik
+        initialValues={{ title: "", release_date: "", poster_path: "", overview: "", runtime: ""}}
+        onSubmit={(values) => {
+          const fromValues = values;
+          fromValues.runtime = +fromValues.runtime;
+          updateMovies(fromValues);
+          isMovieActionPopup(!isMovieAction);
+        }}
+
+        validationSchema={Yup.object().shape({
+          title: Yup.string()
+            .required("Required"),
+          poster_path:  Yup.string()
+            .required("Required"),
+          overview:  Yup.string()
+            .required("Required"),
+          release_date:  Yup.string()
+            .required("Required"),
+          runtime:  Yup.number()
+            .required("Required"),
+        })}
+      >
+        {props => {
+          const {
+            values,
+            errors,
+            isSubmitting,
+            handleChange,
+            handleBlur,
+            handleSubmit,
+          } = props;
+          return (
+            <form onSubmit={handleSubmit}>
+              <label htmlFor="title" className="input-label">
+                <span className="modal-label-text">Title</span>
+                <input
```

Comment:
<Field as="input" ... /> will be better: in this case you won;t need to pass `onChange`, `onBlur`, `value` props

Reply:
Fixed